x/mongo/driver/topology/rtt_monitor.go

```diff
@@ -190,6 +195,31 @@ func min(samples []time.Duration, minSamples int) time.Duration {
 	return min
 }
 
+// percentile90 returns the 90th percentile value of the slice of duration samples. Zero values
+// are not considered samples and are ignored. If no samples or fewer than minSamples are found
+// in the slice, percentile90 returns 0.
+func percentile90(samples []time.Duration, minSamples int) time.Duration {
```

Comment:
_Not_ super confident in my implementation here. I also use [`montanaflynn/stats`](https://github.com/montanaflynn/stats), which was previously only used in our `benchmark` package. I _think_ this will add an extra dependency for users, so let me know if I should just copy-paste the logic instead.

Reply:
Looks great thank you for the code! Will add as part of GODRIVER-2405. I agree that it seems like `montanaflynn/stats` is not an extra dependency.